### ISSUE TITLE
Modeline support?

### ISSUE THREAD

**Issue by ev0rtex** [guillermooo/Six#12]
Is modeline support planned for implementation in Six? I really like being able to specify my syntax in a modeline on a per-file basis like I could before with Vintageous. If development of this feature is being worked on I'd also like to say that it would be awesome if it would pick out the syntax from a standard `# vi set ft=python :` type of modeline.

I have typically had comments such as this in a lot of my files:
```py
# -*- mode: python -*-
# vi: set ft=python :
# sublime: x_syntax Packages/Python/Python.sublime-syntax
```
...it would be awesome if I could remove that last line so that I only have to worry about Emacs and Vim modelines. For now I ended up installing the [Emacs-like Modelines](https://github.com/kvs/STEmacsModelines) plugin which works well with what I already have in my modelines. Ultimately I'd like to be able to remove that plugin and have support in Six.

**Comment by guillermooo** [comment]
Likely, but not a priority right now.

**Comment by ev0rtex** [comment]
Okay, thanks for responding. I wish I could help ;)

**Comment by guillermooo** [comment]
Probably you'll be able to. I think that kind of functionality would be best implemented as a plugin. In this case, you would simply need to (1) be notified on buffer load and (2) have the ability to mutate settings. Proper support for handling settings from Six isn't coming soon, but a basic api for plugins like modelines may be easier to release faster.

**Comment by ev0rtex** [comment]
That would be awesome! When you get a basic plugin API in place I'll work on building this.

I imagine that plugins like this would just be contributed as separate packages that could potentially be installed via Package Control, right?

**Comment by guillermooo** [comment]
Yes, but there would probably be a *Six_Plugin_Core* package that would be a dependency. Modelines would be included in it.
